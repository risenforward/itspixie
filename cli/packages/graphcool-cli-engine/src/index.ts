export { Cluster } from './Cluster'

export { readDefinition } from './GraphcoolDefinition/yaml'
export {
  GraphcoolDefinitionClass,
} from './GraphcoolDefinition/GraphcoolDefinition'

export { EnvDoesntExistError } from './errors/EnvDoesntExistError'

export { Config } from './Config'
export { Command } from './Command'
export { Environment } from './Environment'
export { CLI, run } from './CLI'
export { Output } from './Output'
export {
  RunOptions,
  ProjectDefinition,
  ProjectInfo,
  GraphcoolModule,
  RemoteProject,
  APIError,
  AuthServer,
  AuthTrigger,
  CheckAuth,
  MigrateProjectPayload,
  MigrationActionType,
  MigrationErrorMessage,
  MigrationResult,
  Project,
  Region,
  SchemaInfo,
  PAT,
  AuthenticateCustomerPayload,
  ExternalFiles,
  ExternalFile,
} from './types/common'

export { ClusterConfig, Clusters, RC } from './types/rc'

export { flags, Flag, Arg, Flags } from './Flags'

export { mockDefinition, mockEnv } from './mock'

export { Client } from './Client/Client'
