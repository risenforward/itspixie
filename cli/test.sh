#!/bin/bash

set -e

cd packages/prisma-cli-engine
npm run build
npm test
cd ../../packages/prisma-cli-core
npm test
cd ../../packages/prisma-yml
npm test
