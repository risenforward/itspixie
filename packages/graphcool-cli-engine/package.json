{
  "name": "graphcool-cli-engine",
  "version": "0.0.6",
  "main": "dist/index.js",
  "files": [
    "dist"
  ],
  "typings": "dist/index.d.ts",
  "author": "Tim Suchanek <tim@graph.cool>",
  "license": "MIT",
  "devDependencies": {
    "@types/jest": "^20.0.8",
    "@types/node": "^8.0.22",
    "husky": "^0.14.3",
    "jest": "^21.0.1",
    "lint-staged": "^4.0.3",
    "nock": "^9.0.14",
    "prettier": "^1.5.3",
    "rimraf": "^2.6.1",
    "source-map-support": "^0.4.15",
    "ts-jest": "^21.0.0",
    "tslint": "^5.6.0",
    "tslint-config-prettier": "^1.3.0",
    "tslint-eslint-rules": "^4.1.1",
    "typescript": "^2.4.2"
  },
  "scripts": {
    "build": "rimraf dist && tsc -d",
    "lint": "tslint src/**/*.ts",
    "precommit": "lint-staged",
    "prettier": "prettier --single-quote --no-semi --trailing-comma all --write '*.ts' 'src/**/*.ts'",
    "prepublishOnly": "npm run lint && npm run build",
    "test": "jest && yarn lint"
  },
  "lint-staged": {
    "*.ts": [
      "prettier --single-quote --no-semi --trailing-comma all --write",
      "tslint",
      "git add"
    ]
  },
  "jest": {
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js",
      "jsx",
      "json"
    ],
    "rootDir": "./src",
    "transform": {
      "^.+\\.(ts|tsx)$": "../../../node_modules/ts-jest/preprocessor.js"
    },
    "testMatch": [
      "**/*.test.(ts|js)"
    ],
    "setupFiles": [
      "../test/init.js"
    ]
  },
  "dependencies": {
    "@heroku/linewrap": "^1.0.0",
    "ansi-escapes": "^3.0.0",
    "chalk": "^2.1.0",
    "debug": "^3.0.1",
    "find-up": "^2.1.0",
    "fs-extra": "^4.0.1",
    "klaw-sync": "^3.0.0",
    "lodash.ary": "^4.1.1",
    "lodash.defaults": "^4.2.0",
    "lodash.flatten": "^4.4.0",
    "lodash.get": "^4.4.2",
    "lodash.identity": "^3.0.0",
    "lodash.keys": "^4.2.0",
    "lodash.maxby": "4.x",
    "lodash.merge": "4.x",
    "lodash.partial": "^4.2.1",
    "lodash.property": "^4.4.2",
    "lodash.result": "^4.5.2",
    "lodash.uniqby": "^4.7.0",
    "rwlockfile": "^1.4.8",
    "string": "3.x",
    "string-similarity": "^1.2.0",
    "strip-ansi": "^4.0.0",
    "supports-color": "^4.4.0"
  }
}
