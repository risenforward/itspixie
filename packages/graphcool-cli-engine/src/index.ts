export { EnvDoesntExistError } from './errors/EnvDoesntExistError'

export { Config } from './Config'
export { Command } from './Command'
export { CLI, run } from './CLI'
export { Output } from './Output'
export { ProjectDefinitionClass } from './ProjectDefinition/ProjectDefinition'
export {
  RunOptions,
  ProjectDefinition,
  ProjectInfo,
  GraphcoolModule,
  EnvironmentConfig,
  Environments,
  RemoteProject,
  APIError,
  AuthServer,
  AuthTrigger,
  CheckAuth,
  MigrateProjectPayload,
  MigrationActionType,
  MigrationErrorMessage,
  MigrationResult,
  Project,
  Region,
  SchemaInfo,
} from './types'

export { flags, Flag, Arg, Flags } from './Flags'

