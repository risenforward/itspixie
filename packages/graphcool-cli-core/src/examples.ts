/* tslint:disable */
import { ProjectDefinition } from 'graphcool-cli-engine'

export const defaultDefinition: ProjectDefinition = {
  modules: [
    {
      name: '',
      content: `\
# Welcome to Graphcool!
#
# This file is the main config file for your Graphcool Project.
# It's very minimal at this point and uses default values.
# We've included a hello world function here.
# Just uncomment it an run \`graphcool deploy\`
#
# For full config options, check the docs:
#    graph.cool/docs
#
# Happy Coding!


# GraphQL types
types: ./types.graphql


# uncomment this:

# functions:
#   hello:
#     handler:
#       code:
#         src: ./code/hello.js
#     type: resolver
#     schema: ./schemas/hello.graphql

 
# Graphcool modules
modules: {}


# Model/Relation permissions
permissions:
- operation: "*"

  
# Permanent Auth Token / Root Tokens
rootTokens: []

`,
      files: {
        './types.graphql': `\
# This file contains the GraphQL Types
# Graphcool has one special type, the File type:

# type File implements Node {
#   contentType: String!
#   createdAt: DateTime!
#   id: ID! @isUnique
#   name: String!
#   secret: String! @isUnique
#   size: Int!
#   updatedAt: DateTime!
#   url: String! @isUnique
# }

# All types need to have an id field defined like this:

# type User implements Node {
#   id: ID! @isUnique
# }

`,
        './code/hello.js': `\
module.exports = event => {
  return {
    data: {
      message: \`Hello $\{event.data.name || 'World'\}\`
    }
  }
}`,
        './schemas/hello.graphql': `\
type HelloPayload {
  message: String!
}

extend type Query {
  hello(name: String): HelloPayload
}
`,
      },
    },
  ],
}

export const changedDefaultDefinition: ProjectDefinition = {
  modules: [
    {
      name: '',
      content: `
types: ./types.graphql
functions: {}
permissions:
- isEnabled: true
  operation: File.read
  authenticated: false
- isEnabled: true
  operation: File.create
  authenticated: false
- isEnabled: true
  operation: File.update
  authenticated: false
- isEnabled: true
  operation: File.delete
  authenticated: false
- isEnabled: true
  operation: User.read
  authenticated: false
- isEnabled: true
  operation: User.create
  authenticated: false
- isEnabled: true
  operation: User.update
  authenticated: false
- isEnabled: true
  operation: User.delete
  authenticated: false
rootTokens: []
`,
      files: {
        './types.graphql': `type File implements Node {
  contentType: String!
  createdAt: DateTime!
  id: ID! @isUnique
  name: String!
  secret: String! @isUnique
  size: Int!
  updatedAt: DateTime!
  url: String! @isUnique
}

type User implements Node {
  createdAt: DateTime!
  id: ID! @isUnique
  updatedAt: DateTime!
}

type Post {
  imageUrl: String!
}
`,
      },
    },
  ],
}

export const advancedDefinition: ProjectDefinition = {
  modules: [
    {
      name: '',
      content:
        'types: ./types.graphql\nfunctions:\n  filter-posts:\n    handler:\n      code:\n        src: ./code/filter-posts.js\n    type: operationBefore\n    operation: Post.create\n  log-posts:\n    handler:\n      code:\n        src: ./code/log-posts.js\n    type: subscription\n    query: ./code/log-posts.graphql\n  weather:\n    handler:\n      code:\n        src: ./code/weather.js\n    type: resolver\n    schema: ./code/weather.graphql\npermissions:\n- operation: File.read\n- operation: File.create\n- operation: File.update\n- operation: File.delete\n- operation: Post.read\n- operation: Post.update\n- operation: Post.delete\n- operation: Post.create\n  authenticated: true\n- operation: User.read\n- operation: User.create\n- operation: User.update\n- operation: User.delete\nrootTokens: []\n',
      files: {
        './types.graphql':
          'type File implements Node {\n  contentType: String!\n  createdAt: DateTime!\n  id: ID! @isUnique\n  name: String!\n  secret: String! @isUnique\n  size: Int!\n  updatedAt: DateTime!\n  url: String! @isUnique\n}\n\ntype User implements Node {\n  createdAt: DateTime!\n  id: ID! @isUnique\n  updatedAt: DateTime!\n}\n\ntype Post implements Node {\n  title: String!\n  description: String!\n  createdAt: DateTime!\n  id: ID! @isUnique\n  updatedAt: DateTime!\n}',
        './code/log-posts.js':
          '// Click "EXAMPLE EVENT" to see whats in `event`\nmodule.exports = function (event) {\n  console.log(event.data)\n  return {data: event.data}\n}\n',
        './code/log-posts.graphql':
          'subscription {\n  Post(filter: {\n    mutation_in: [CREATED, UPDATED, DELETED]\n  }) {\n    updatedFields\n    node {\n      id\n    }\n  }\n}',
        './code/filter-posts.js':
          "// Click \"EXAMPLE EVENT\" to see whats in `event`\nmodule.exports = function (event) {\n  console.log(event.data)\n  if (event.data.createPost.description.includes('bad') {\n  \treturn {error: 'bad is not allowed'}\n  }\n  return {data: event.data}\n}\n",
        './code/weather.js':
          "const fetch = require('node-fetch')\n\nmodule.exports = function (event) {\n  const city = event.data.city\n  return fetch(getApiUrl(city))\n  .then(res => res.json())\n  .then(data => {\n    console.log(data)\n    return {\n      data: {\n        temperature: data.main.temp,\n        pressure: data.main.pressure,\n        windSpeed: data.wind.speed,\n      }\n    }\n  })\n}\n\nfunction getApiUrl(query) {\n\treturn `http://samples.openweathermap.org/data/2.5/weather?q=${query}&appid=b1b15e88fa797225412429c1c50c122a1`\n  }",
        './code/weather.graphql':
          'type WeatherPayload {\n  temperature: Float!\n  pressure: Float!\n  windSpeed: Float!\n}\n\nextend type Query {\n  weather(city: String!): WeatherPayload\n}\n',
      },
    },
  ],
}

export const examples = {
  blank: defaultDefinition,
  instagram: advancedDefinition,
  stripe: advancedDefinition,
  sendgrid: advancedDefinition,
}
