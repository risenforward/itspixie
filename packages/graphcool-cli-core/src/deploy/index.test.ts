import * as nock from 'nock'
import {Config} from 'graphcool-cli-engine'
import { init } from '../test/mock-requests'
import Deploy from './'
import { changedDefaultDefinition, defaultDefinition } from '../examples'

const config = new Config()
describe('deploy', () => {
  test('without change', async () => {

    nock('https://dev.api.graph.cool:443', {"encodedQueryParams":true})
      .post('/system', {"query":"{\n      viewer {\n        user {\n          id\n          email\n        }\n      }\n    }"})
      .reply(200, {"data":{"viewer":{"user":{"id":"cj7ektagf09sw0112cxeyoqjr","email":"test-ci@graph.cool"}}}}, [ 'Content-Type',
        'application/json',
        'Content-Length',
        '92',
        'Connection',
        'close',
        'Date',
        'Sun, 10 Sep 2017 11:00:06 GMT',
        'Request-Id',
        'eu-west-1:system:cj7emob0n0ai50112657j1x3h',
        'Server',
        'akka-http/10.0.8',
        'X-Cache',
        'Miss from cloudfront',
        'Via',
        '1.1 926c5f53581f4e2717deb4e0fac4efc6.cloudfront.net (CloudFront)',
        'X-Amz-Cf-Id',
        '8ld-M4A-gszxxWbO0fr2LipWQf4Pa6fuwcTteK_z0lwn0kIkOP5bgQ==' ]);


    nock('https://dev.api.graph.cool:443', {"encodedQueryParams":true})
      .post('/system', {"query":"      mutation($projectId: String!, $force: Boolean, $isDryRun: Boolean!, $config: String!) {\n        push(input: {\n          projectId: $projectId\n          force: $force\n          isDryRun: $isDryRun\n          config: $config\n          version: 1\n        }) {\n          migrationMessages {\n            type\n            action\n            name\n            description\n            subDescriptions {\n              type\n              action\n              name\n              description\n            }\n          }\n          errors {\n            description\n            type\n            field\n          }\n          project {\n            id\n            name\n            alias\n            projectDefinitionWithFileContent\n          }\n        }\n      }\n    ","variables":{"projectId":"cj7elbt9z0a3z011214m21uwn","isDryRun":false,"config":"{\"modules\":[{\"name\":\"\",\"content\":\"\\ntypes: ./types.graphql\\nfunctions: {}\\npermissions:\\n- isEnabled: true\\n  operation: File.read\\n  authenticated: false\\n- isEnabled: true\\n  operation: File.create\\n  authenticated: false\\n- isEnabled: true\\n  operation: File.update\\n  authenticated: false\\n- isEnabled: true\\n  operation: File.delete\\n  authenticated: false\\n- isEnabled: true\\n  operation: User.read\\n  authenticated: false\\n- isEnabled: true\\n  operation: User.create\\n  authenticated: false\\n- isEnabled: true\\n  operation: User.update\\n  authenticated: false\\n- isEnabled: true\\n  operation: User.delete\\n  authenticated: false\\nrootTokens: []\\n\",\"files\":{\"./types.graphql\":\"type File implements Node {\\n  contentType: String!\\n  createdAt: DateTime!\\n  id: ID! @isUnique\\n  name: String!\\n  secret: String! @isUnique\\n  size: Int!\\n  updatedAt: DateTime!\\n  url: String! @isUnique\\n}\\n\\ntype User implements Node {\\n  createdAt: DateTime!\\n  id: ID! @isUnique\\n  updatedAt: DateTime!\\n}\\n\"}}]}"}})
      .reply(200, {"data":{"push":{"migrationMessages":[],"errors":[],"project":{"id":"cj7elbt9z0a3z011214m21uwn","name":"Shellcrest Cub","alias":null,"projectDefinitionWithFileContent":"{\n  \"modules\": [{\n    \"name\": \"\",\n    \"content\": \"types: ./types.graphql\\nfunctions: {}\\npermissions:\\n- operation: File.read\\n- operation: File.create\\n- operation: File.update\\n- operation: File.delete\\n- operation: User.read\\n- operation: User.create\\n- operation: User.update\\n- operation: User.delete\\nrootTokens: []\\n\",\n    \"files\": {\n      \"./types.graphql\": \"type File implements Node {\\n  contentType: String!\\n  createdAt: DateTime!\\n  id: ID! @isUnique\\n  name: String!\\n  secret: String! @isUnique\\n  size: Int!\\n  updatedAt: DateTime!\\n  url: String! @isUnique\\n}\\n\\ntype User implements Node {\\n  createdAt: DateTime!\\n  id: ID! @isUnique\\n  updatedAt: DateTime!\\n}\"\n    }\n  }]\n}"}}}}, [ 'Content-Type',
        'application/json',
        'Content-Length',
        '918',
        'Connection',
        'close',
        'Date',
        'Sun, 10 Sep 2017 11:00:07 GMT',
        'Request-Id',
        'eu-west-1:system:cj7emobp90ai60112qe9jo689',
        'Server',
        'akka-http/10.0.8',
        'X-Cache',
        'Miss from cloudfront',
        'Via',
        '1.1 b451ce1932d9b97c4ef54f2f37ecb931.cloudfront.net (CloudFront)',
        'X-Amz-Cf-Id',
        '-fkEiv0LsvSXlLfLDXUPYv6duhBRFbQpsxETOP8aVW6NXQzk78xI6A==' ]);

    Deploy.mockDefinition = defaultDefinition
    Deploy.mockEnv = {
      default: 'dev',
      environments: {
        dev: 'cj7elbt9z0a3z011214m21uwn'
      }
    }
    const result = await Deploy.mock()
    expect(result.out.stdout.output).toMatchSnapshot()

    nock.cleanAll()
  })

  test('with change', async () => {

    nock('https://dev.api.graph.cool:443', {"encodedQueryParams":true})
      .post('/system', {"query":"{\n      viewer {\n        user {\n          id\n          email\n        }\n      }\n    }"})
      .reply(200, {"data":{"viewer":{"user":{"id":"cj7ektagf09sw0112cxeyoqjr","email":"test-ci@graph.cool"}}}}, [ 'Content-Type',
        'application/json',
        'Content-Length',
        '92',
        'Connection',
        'close',
        'Date',
        'Sun, 10 Sep 2017 11:11:27 GMT',
        'Request-Id',
        'eu-west-1:system:cj7en2wxc0am20112d9ldfa02',
        'Server',
        'akka-http/10.0.8',
        'X-Cache',
        'Miss from cloudfront',
        'Via',
        '1.1 e482e2c19d6e57adc72e19f731c7bf44.cloudfront.net (CloudFront)',
        'X-Amz-Cf-Id',
        'dVPKelegAj38sDAcZ8cVuTCgWbNgB27_dG7zv9RUO5enmwXE9wbYhg==' ]);


    nock('https://dev.api.graph.cool:443', {"encodedQueryParams":true})
      .post('/system', {"query":"      mutation($projectId: String!, $force: Boolean, $isDryRun: Boolean!, $config: String!) {\n        push(input: {\n          projectId: $projectId\n          force: $force\n          isDryRun: $isDryRun\n          config: $config\n          version: 1\n        }) {\n          migrationMessages {\n            type\n            action\n            name\n            description\n            subDescriptions {\n              type\n              action\n              name\n              description\n            }\n          }\n          errors {\n            description\n            type\n            field\n          }\n          project {\n            id\n            name\n            alias\n            projectDefinitionWithFileContent\n          }\n        }\n      }\n    ","variables":{"projectId":"cj7elbt9z0a3z011214m21uwn","isDryRun":false,"config":"{\"modules\":[{\"name\":\"\",\"content\":\"\\ntypes: ./types.graphql\\nfunctions: {}\\npermissions:\\n- isEnabled: true\\n  operation: File.read\\n  authenticated: false\\n- isEnabled: true\\n  operation: File.create\\n  authenticated: false\\n- isEnabled: true\\n  operation: File.update\\n  authenticated: false\\n- isEnabled: true\\n  operation: File.delete\\n  authenticated: false\\n- isEnabled: true\\n  operation: User.read\\n  authenticated: false\\n- isEnabled: true\\n  operation: User.create\\n  authenticated: false\\n- isEnabled: true\\n  operation: User.update\\n  authenticated: false\\n- isEnabled: true\\n  operation: User.delete\\n  authenticated: false\\nrootTokens: []\\n\",\"files\":{\"./types.graphql\":\"type File implements Node {\\n  contentType: String!\\n  createdAt: DateTime!\\n  id: ID! @isUnique\\n  name: String!\\n  secret: String! @isUnique\\n  size: Int!\\n  updatedAt: DateTime!\\n  url: String! @isUnique\\n}\\n\\ntype User implements Node {\\n  createdAt: DateTime!\\n  id: ID! @isUnique\\n  updatedAt: DateTime!\\n}\\n\\ntype Post {\\n  imageUrl: String!\\n}\\n\"}}]}"}})
      .reply(200, ["1f8b0800000000000003a454df6bdb3010fe57ae7aceb2a61b8cf96925a1d087964113f61007e25897449d2cb9924c5842fef7dd494edaa6f6c3983148febebbfbee173e08598442640751377ecb67a536ae08ca9a07f4bed8a017d9fc204c51a1c8c44feb83180889be74aa662b026fc1e00ec29f1a61a7c216c21681ed61c9e64b501e4a87454039245fdfac26afee293afb52a03b855a924951b691c7d18d90565e5594d0cce99e14d6ec7f99c3c96709859129cbe55370ca6caedea7765c0c4e894cf9f898075ba073d671d674af9d7dc63270d79424b3f2f91bea55f8bebf2ebeecaf47a39bd1d7ea66d4eccc6b054f5bd49a247d8071b3620dad0a0a671aadcf0127b85646b1f62faae54e691c5b13d0909238e406201795958d469f8b0ce61162903518c9452e0627b04cae09e7fa7c06c3cff132a449d7db179de766dd98582c9187237dd7e82ae57d44e8f3135842e25a64c0090da923b29348fdeca49a5af65112357ea4787f3a9522d1ad14a96ea548752bcd3cba4ea548742b45aa5b29526725676d98dadfc8dd9d2f0878339eb56ac7d84e91b1cbf19c47177b05aaaa355634530f8f562279b26b3b675edd0cda054f445af0db90c1842e5355612294cce07e72053f949f19f5d260447989de07f04821c219bbb0f76a4ff6f72624e3d48f0eb5c6e9ee10bc6ef4c6fab86f7df5fd5b193d79bc15e375e8113bfd34feab93bd298834ea231fc70521e248cf5f000000ffff0300b2e62f5b8b050000"], [ 'Content-Type',
        'application/json',
        'Transfer-Encoding',
        'chunked',
        'Connection',
        'close',
        'Vary',
        'Accept-Encoding',
        'Date',
        'Sun, 10 Sep 2017 11:11:28 GMT',
        'Request-Id',
        'eu-west-1:system:cj7en2x2k0am30112muvlj98k',
        'Server',
        'akka-http/10.0.8',
        'Content-Encoding',
        'gzip',
        'X-Cache',
        'Miss from cloudfront',
        'Via',
        '1.1 926c5f53581f4e2717deb4e0fac4efc6.cloudfront.net (CloudFront)',
        'X-Amz-Cf-Id',
        'FZwO6vcREexdulKnOHzTgTeR9I0R4yY4z85031kkFPOKSZXqDEpfHQ==' ]);

    Deploy.mockDefinition = changedDefaultDefinition
    Deploy.mockEnv = {
      default: 'dev',
      environments: {
        dev: 'cj7elbt9z0a3z011214m21uwn'
      }
    }
    const result = await Deploy.mock()
    expect(result.out.stdout.output).toMatchSnapshot()

    nock.cleanAll()
  })

})