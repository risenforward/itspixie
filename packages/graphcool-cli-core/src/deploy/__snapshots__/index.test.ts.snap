// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deploy with change 1`] = `
"✔ Your project cj7elbt9z0a3z011214m21uwn of env dev was successfully updated.
            Here are the changes: 


  | (+)  A new type with the name \`Post\` is created.

  └── (+)  A new field with the name \`imageUrl\` and type \`String!\` is created.

"
`;

exports[`deploy without change 1`] = `
"✔ Identical project definition for project cj7elbt9z0a3z011214m21uwn in env dev, no action required.
"
`;
